test/extended/builds/valuefrom.go

```diff
@@ -0,0 +1,126 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+	})
+
+	g.It("should successfully resolve valueFrom in s2i build environment variables", func() {
+
+		g.By("creating test successful build config")
+		err := oc.Run("create").Args("-f", successfulSTIBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, err := exutil.StartBuildAndWait(oc, "mys2itest")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		br.AssertSuccess()
+
+		logs, _ := br.Logs()
+		o.Expect(logs).To(o.ContainSubstring("FIELDREF_ENV=mys2itest-1"))
+		o.Expect(logs).To(o.ContainSubstring("CONFIGMAPKEYREF_ENV=myvalue"))
+		o.Expect(logs).To(o.ContainSubstring("SECRETKEYREF_ENV=developer"))
+		o.Expect(logs).To(o.ContainSubstring("FIELDREF_CLONE_ENV=mys2itest-1"))
+		o.Expect(logs).To(o.ContainSubstring("FIELDREF_CLONE_CLONE_ENV=mys2itest-1"))
+		o.Expect(logs).To(o.ContainSubstring("UNAVAILABLE_ENV=$(SOME_OTHER_ENV"))
+		o.Expect(logs).To(o.ContainSubstring("ESCAPED_ENV=$(MY_ESCAPED_VALUE)"))
+
+	})
+
+	g.It("should successfully resolve valueFrom in docker build environment variables", func() {
+
+		g.By("creating test successful build config")
+		err := oc.Run("create").Args("-f", successfulDockerBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, err := exutil.StartBuildAndWait(oc, "mydockertest")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		br.AssertSuccess()
+
+		logs, _ := br.Logs()
+		o.Expect(logs).To(o.ContainSubstring("\"FIELDREF_ENV\" \"mydockertest-1\""))
+		o.Expect(logs).To(o.ContainSubstring("\"CONFIGMAPKEYREF_ENV\" \"myvalue\""))
+		o.Expect(logs).To(o.ContainSubstring("\"SECRETKEYREF_ENV\" \"developer\""))
+		o.Expect(logs).To(o.ContainSubstring("\"FIELDREF_CLONE_ENV\" \"mydockertest-1\""))
+		o.Expect(logs).To(o.ContainSubstring("\"FIELDREF_CLONE_CLONE_ENV\" \"mydockertest-1\""))
+		o.Expect(logs).To(o.ContainSubstring("\"UNAVAILABLE_ENV\" \"$(SOME_OTHER_ENV)\""))
+		o.Expect(logs).To(o.ContainSubstring("\"ESCAPED_ENV\" \"$(MY_ESCAPED_VALUE)\""))
+
+	})
+
+	g.It("should fail resolving unresolveable valueFrom in sti build environment variable references", func() {
+
+		g.By("creating test build config")
+		err := oc.Run("create").Args("-f", failedSTIBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, _ := exutil.StartBuildAndWait(oc, "mys2itest")
+		br.AssertFailure()
+
+		o.Expect(strings.Contains(fmt.Sprintf("%#v", br.Build.Status.Reason), "UnresolvableEnvironmentVariable")).To(o.BeTrue())
+		o.Expect(strings.Contains(fmt.Sprintf("%#v", br.Build.Status.Message), "unsupported fieldPath: metadata.nofield")).To(o.BeTrue())
+		o.Expect(strings.Contains(fmt.Sprintf("%#v", br.Build.Status.Message), "key nokey not found in config map myconfigmap")).To(o.BeTrue())
+		o.Expect(strings.Contains(fmt.Sprintf("%#v", br.Build.Status.Message), "key nousername not found in secret mysecret")).To(o.BeTrue())
```

Comment:
also i'm not seeing how you're getting these specific messages into the build status message, when you're only setting it as a constant value above?